root/docker/docker-entrypoint

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
+
+USER_NAME=${USER_NAME:-gatling-user}
+echo "User name is set to $USER_NAME"
+
+PUID=${PUID:-911}
+PGID=${PGID:-911}
+
+if [[ "$USER_NAME" != "gatling-user" ]]; then
+  usermod -l "$USER_NAME" gatling-user
+  groupmod -n "$USER_NAME" gatling-user
+fi
+
+groupmod -o -g "$PGID" "$USER_NAME"
+usermod -o -u "$PUID" "$USER_NAME"
+
+echo "
+───────────────────────────────────────
+GID/UID
+───────────────────────────────────────
+User UID:    $(id -u "${USER_NAME}")
+User GID:    $(id -g "${USER_NAME}")
+───────────────────────────────────────
+"
+
+# create folders
+mkdir -p \
+    /config/{.ssh,ssh_host_keys,logs/openssh} \
+    /run/sshd
+
+# symlink out ssh config directory
+if [[ ! -L /etc/ssh ]]; then
+    if [[ ! -f /config/ssh_host_keys/sshd_config ]]; then
+        sed -i '/#PidFile/c\PidFile \/config\/sshd.pid' /etc/ssh/sshd_config
+        cp -a /etc/ssh/sshd_config /config/ssh_host_keys/
+    fi
+    rm -Rf /etc/ssh
+    ln -s /config/ssh_host_keys /etc/ssh
+    ssh-keygen -A
+fi
+
+# set umask for sftp
+UMASK=${UMASK:-022}
+sed -i "s|/usr/lib/ssh/sftp-server$|/usr/lib/ssh/sftp-server -u ${UMASK}|g" /etc/ssh/sshd_config
```

Comment:
What's the default configuration? Where come from the need to modify it ?

Reply:
As far as I understand:
- OpenSSH's default is to simply use the user's default umask
- `-u` forces a specific umask instead, for all new files/directories created, regardless of the user doing it

Reply:
ensure that the connected user can create AND modify the file they upload